app/queries/link_set_schema.rb

```diff
@@ -0,0 +1,55 @@
+module Queries
+  class LinkSetSchema
+    def call
+      default_link_types.merge(topical_events)
+    end
+
+  private
+
+    def default_link_types
+      (recursive_link_types + non_recursive_link_types).each_with_object({}) do |field, hash|
+        hash[field] = {
+          expanded_links: default_links,
+          recurse: recursive_link_types.include?(field)
+        }
+      end
+    end
+
+    def recursive_link_types
+      [:active_top_level_browse_page, :alpha_taxons, :can_be_replaced_by,
+       :children, :content_owners, :document_collections, :email_alert_signup,
+       :lead_organisations, :link_type, :linked_items,
+       :mainstream_browse_pages, :manual, :parent, :people]
+    end
+
+    def non_recursive_link_types
+      [:organisations, :policy_areas, :press_releases, :related,
+       :related_policies, :related_topics, :second_level_browse_pages,
+       :sections, :supporting_organisations, :top_level_browse_pages,
+       :topics, :working_groups, :world_locations, :worldwide_organisations,
+       :worldwide_priorities]
+    end
+
+    def topical_events
+      {
+        topical_event: {
+          expanded_links: default_links + [:details],
+          recurse: false
+        }
+      }
+    end
+
+    def default_links
+      [
+        :analytics_identifier,
+        :api_url,
+        :base_path,
+        :content_id,
+        :description,
+        :locale,
+        :title,
+        :web_url
+      ]
+    end
+  end
+end
```

Comment:
yes this is the same thing. This is just a rename of that file, but both use this to compute and to update.
It may be a bit too far with the reducing the repetition since I am still not certain what the file will look like in the end.

Also this is arbitrarily split, I am not sure how we will decide which link_types will be recursive or not?